<launch>
	<!-- global rosparam -->
	<param name="use_sim_time" value="true"/>
	<arg name="bagname" default="2020-10-07-08-29-57"/>

	<!-- rosbag -->
	<node pkg="rosbag" type="play" name="player" args="--clock /home/amsl/ozaki/bagfiles/stick/campus/$(arg bagname).bag"/>

	<!-- rviz -->
	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find save_dataset)/rviz_config/save_still_imu_camera.rviz"/>

	<!-- gyrodometry -->
	<node pkg="gyrodometry" name="gyrodometry3d" type="gyrodometry3d">
		<param name="wait_inipose" type="bool" value="false"/>
	</node>

	<!-- main -->
	<node pkg="save_dataset" type="odom_check_diff_and_still" name="odom_check_diff_and_still" output="screen">
		<remap from="/odom" to="/gyrodometry"/>
		<param name="th_diff_position_m" type="double" value="0.5"/>
		<param name="th_diff_angle_deg" type="double" value="5.0"/>
	</node>
</launch>
